File: app/src/main/java/com/llj/architecturedemo/MainActivityModule.java
package com.llj.architecturedemo;

import android.arch.lifecycle.ViewModelProviders;

import dagger.Module;
import dagger.Provides;

/**
 * ArchitectureDemo
 * describe:
 * author liulj
 * date 2018/5/16
 */
@Module
public class MainActivityModule {
    @Provides
    MainContract.View provideView(MainActivity activity) {
        return activity;
    }

    @Provides
    MainContract.ViewModel provideModel(MainActivity activity) {
        return ViewModelProviders.of(activity).get(MainContractViewModel.class);
    }

}

File: app/src/p_app/main/java/com/llj/architecturedemo/ui/module/MainActivityModule.kt
package com.llj.architecturedemo.ui.module

import com.llj.architecturedemo.ui.activity.MainActivity
import com.llj.architecturedemo.ui.view.MainContractView

import dagger.Module
import dagger.Provides

/**
 * ArchitectureDemo
 * describe:
 * @author liulj
 * @date 2018/5/16
 */
@Module
class MainActivityModule {
  @Provides
  internal fun provideView(activity: MainActivity): MainContractView {
    return activity
  }
}
